src/main/asciidoc/_chapters/hbase-default.adoc

```diff
@@ -1949,6 +1949,34 @@ If the DFSClient configuration
 `simple`
 
 
+[[hbase.security.authentication.ui.metrics.protected]]
+*`hbase.security.authentication.ui.metrics.protected`*::
++
+.Description
+
+      Controls whether or not metrics endpoints are allowed only for admin.
+      If true, only users listed on "hbase.security.authentication.spnego.admin.users"
+      or users in group listed on "hbase.security.authentication.spnego.admin.groups"
+      are allowed to access metrics endpoints. (e.g. /jmx, /metrics, /prometheus)
+
++
+.Default
+`false`
+
+
+[[hbase.security.authentication.spnego.kerberos.endpoint.whitelist]]
```

Comment:
Was going through hadoop PR. This property has been implemented in hadoop to handle any endpoint irrespective of whether it is metrics or something else. Why don't we follow same approach.? Do we really need to handle just metrics endpoint and have the other property? A generic implementation like the one in hadoop sound more useful to me. Also the name of the config will confuse others. As it seems like a generic whitelist.

Reply:
If you want to use the name "hadoop.http.authentication.kerberos.endpoint.whitelist", instead of initializing params using configurations that starts with `HTTP_SPNEGO_AUTHENTICATION_PREFIX`, that name should be used like:

```java
String endpointWhitelist = getOrEmptyString(conf, "hadoop.http.authentication.kerberos.endpoint.whitelist");
if (!endpointWhitelist.isEmpty()) {
  params.put("endpoint.whitelist", endpointWhitelist);
}
```
Do you think it is more suitable for HBase code?

Reply:
> If you want to use the name "hadoop.http.authentication.kerberos.endpoint.whitelist"

Hey I did not mean to change name. 

> A generic implementation like the one in hadoop sound more useful to me.

What I am trying to say is current implementation in PR aims to just whitelist which are under metrics umbrella. But if you look at the hadoop PR, it aims to provide a way to whitelist any endpoint in the passed list. Can we not follow that approach and have a generic implementation where we let the user whitelist any endpoint irrespective of whether it is a metric endpoint or say /logs

>  Also the name of the config will confuse others. As it seems like a generic whitelist. 

I mean nowhere in config name `hbase.security.authentication.spnego.kerberos.endpoint.whitelist` it is obvious that this can control only metrics endpoints.

Reply:
Ignore as my doubt is clear here. Just that I think even conf servlet would also require the whitelisting if one want to access it: https://github.com/apache/hbase/blob/8d8b05221bb3d07eb48058747359406c3be951a8/hbase-http/src/main/java/org/apache/hadoop/hbase/http/HttpServer.java#L827? 

So is this documentation correct?

Reply:
Yes, if someone wants to disable spnego authentication completely on `/conf` endpoint completely, `/conf` should also be added to whitelist.

Reply:
Maybe then update docs to reflect same?

Reply:
I updated docs.